Retrovizor.Domain/Repositories/Implementations/ExamRepository.cs

```diff
@@ -0,0 +1,97 @@
+﻿using Retrovizor.Data.Entities;
+using Retrovizor.Data.Entities.Models;
+using Retrovizor.Domain.Repositories.Interfaces;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Retrovizor.Domain.Repositories.Implementations
+{
+    public class ExamRepository : IExamRepository
+    {
+        public ExamRepository(RetrovizorContext context)
+        {
+            _context = context;
+        }
+        private readonly RetrovizorContext _context;
+
+        public List<Exam> GetAllExamsByDrivingSchoolId(int id)
+        {
+            var students = _context.Students.Where(s => s.DrivingSchoolId == id);
+
+            var studentExams = new List<StudentExam>();
+
+            foreach(var student in students)
+                studentExams.AddRange(student.StudentExams);
+
+            var exams = new List<Exam>();
+
+            foreach(var studentExam in studentExams)
+                exams.Add(studentExam.Exam);
+
+            return exams.Distinct().ToList();
+
+        }
+        public bool AddExam(Exam examToAdd)
+        {
+            var doesExamExist = _context.Exams.Any(e => string.Equals(e.Type, examToAdd.Type, StringComparison.CurrentCultureIgnoreCase));
+
+            if(doesExamExist)
+                return false;
+
+            _context.Exams.Add(examToAdd);
+            _context.SaveChanges();
+            return true;
+        }
+
+        public bool EditExam(Exam editedExam)
+        {
+            var doesExamExist = _context.Exams.Any(e => string.Equals(e.Type, editedExam.Type, StringComparison.CurrentCultureIgnoreCase));
```

Comment:
nemos editat exam ako mu se ne promini type. Ako zelis prominit total points ne mozes jer moras uz to prominit i type

Reply:
doda san  "&& editedExam.Id != e.Id" u uvjet argumenta za .Any , triba pazit da se ne mogu dva exam typea nameat isto. sad ce prolazit i za nepromijenjeni type